File: src/SIL.XForge.Scripture/ClientApp/src/app/core/models/translate-metrics.ts
export type TranslateMetricsType = 'edit' | 'navigate';

export interface TranslateMetrics {
  id: string;
  type: TranslateMetricsType;
  sessionId: string;
  textRef: string;
  chapter: number;

  // editing metrics
  segment?: string;
  sourceWordCount?: number;
  targetWordCount?: number;
  keyBackspaceCount?: number;
  keyDeleteCount?: number;
  keyCharacterCount?: number;
  productiveCharacterCount?: number;
  suggestionAcceptedCount?: number;
  suggestionTotalCount?: number;
  timeEditActive?: number;

  // navigation metrics
  keyNavigationCount?: number;
  mouseClickCount?: number;
}

File: src/SIL.XForge.Scripture/Models/TranslateMetrics.cs
using SIL.XForge.Models;

namespace SIL.XForge.Scripture.Models
{
    public class TranslateMetrics : Entity
    {
        public string Type { get; set; }
        public string SessionId { get; set; }
        public string UserRef { get; set; }
        public string ProjectRef { get; set; }
        public string TextRef { get; set; }
        public int Chapter { get; set; }

        // navigation metrics
        public int KeyNavigationCount { get; set; }
        public int MouseClickCount { get; set; }

        // editing metrics
        public int KeyBackspaceCount { get; set; }
        public int KeyDeleteCount { get; set; }
        public int KeyCharacterCount { get; set; }
        public int ProductiveCharacterCount { get; set; }
        public int SuggestionAcceptedCount { get; set; }
        public int SuggestionTotalCount { get; set; }
        public int TimeEditActive { get; set; }
    }
}
